DataCapturing/Source/Model/Measurement.swift

```diff
@@ -122,6 +104,58 @@ public class Measurement: Hashable, Equatable {
         self.tracks = tracks
     }
 
+    // MARK: - Public Interface
+    /// The number of accelerations stored by this measurement. This information is required to deserialize the binary format for those values.
+    public var accelerationsCount: Int32
+    /// The number of rotations stored by this measurement. This information is required to deserialize the binary format for those values.
+    public var rotationsCount: Int32
+    /// The number of directions stored by this measurement. This information is required to deserialize the binary format for those values.
+    public var directionsCount: Int32
+    /// A device wide unique identifier for this measurement. Usually set by incrementing a counter.
+    public let identifier: Int64
+    /// A flag, marking this `Measurement` as either ready for data syncrhonization or not.
+    public var synchronizable: Bool
+    /// A flag, marking this `Measurement` as either synchronized or not.
+    public var synchronized: Bool
+    /// The UNIX timestamp in milliseconds since the 1st of January 1970, when this measurement was started.
+    public let timestamp: Int64
+    /// The calculated length of the `Measurement`. See `DistanceCalculationStrategy` for further details.
+    public var trackLength: Double
+    /// The user events that occurred during this `Measurement`.
+    public var events: [Event]
+    /// The tracks containing all the data this `Measurement` has captured.
+    public var tracks: [Track]
+
+    /// Required by the `Hashable` protocol to produce a hash for this object.
+    public func hash(into hasher: inout Hasher) {
+        hasher.combine(identifier)
+    }
+
+    /// Required by the `Equatable` protocol to compare two `Measurement` instances.
+    public static func == (lhs: Measurement, rhs: Measurement) -> Bool {
+        return lhs.identifier == rhs.identifier
+    }
+
+    /// Provide the average speed from the provided measurement
+    public func averageSpeed() -> Double {
+        var sum = 0.0
+        var counter = 0
+        tracks.forEach { track in
+            track.locations.forEach { location in
+                sum += location.speed
+                counter += 1
+            }
+        }
+
+        if counter==0 {
+            return 0.0
+        } else {
+            return sum/Double(counter)
+        }
+    }
```

Comment:
This is the only changed part. Everything else has only been moved around.

Reply:
The only question (in comparision with the Android code) would be:

Didn't we want to use the `LocationCleaningStrategy` in average speed calculation to not count the standing time "speed ~ 0"?

See here: https://github.com/cyface-de/android-backend/pull/257#discussion_r1068095043